parser.c

```diff
@@ -338,14 +338,36 @@ static struct kanshi_profile_output *parse_profile_output(
 	}
 }
 
-static struct kanshi_profile *parse_profile(struct kanshi_parser *parser) {
-	if (!parser_expect_token(parser, KANSHI_TOKEN_LBRACKET)) {
-		return NULL;
+static bool kanshi_parse_profile_name(struct kanshi_parser *parser, char** name) {
```

Comment:
Isn't `parser_read_str` enough?

Reply:
Not too sure about this one, but it seems that using `parser_read_str` would requires a configuration name because `parser_read_str` will not accept an opening `{`. The way I implemented it the name remains optional and the file format backward compatible

Reply:
I believe `parser_read_str` would just return fill `parser->tok_str` with an empty string, returning immediately, if there is no profile name?

Reply:
> I believe `parser_read_str` would just return fill `parser->tok_str` with an empty string, returning immediately, if there is no profile name?

It works indeed. The not-so-nice catch is that you have to reset `tok_str_len` manually. See updated patch.